File: Data Structures & Algorithms/length-of-last-word/submission-1.java
class Solution {
    public int lengthOfLastWord(String s) {
        int ans = 0,curr = 0;
        for(int i = 0; i < s.length(); i++)
        {
            if(s.charAt(i) == ' ') 
            {
                if(curr == 0) continue;
                ans = curr;
                curr = 0;
            }
            else curr++;
        }   
        if(curr > 0) ans = curr;
        return ans;
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-1.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int t) {
        int l = 0;
        int r = nums.size() - 1;
        int ans = -1;
        int m;
        if(nums[r] < t) ans = r + 1;
        else if(nums[l] > t) ans = l;
        else
        {
            while(l <= r)
            {
                m = l + (r - l)/2;
                if(nums[m] == t)
                {
                    ans = m;
                    break;
                }
                else if(m > 0 && nums[m] > t && nums[m - 1] < t)
                {
                    ans = m;
                    break;
                }
                else if(nums[m] > t) r = m - 1;
                else if(nums[m] < t) l = m + 1;
            }
        }
        return ans;
    }
};